plugins/redis/password.go

```diff
@@ -0,0 +1,38 @@
+package redis
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func Password() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.Password,
```

Comment:
Looking at the [Redis authentication docs](https://redis.io/docs/management/security/#authentication), they refer to this option as "legacy authentication". I do agree that we should offer this, but we may also want to offer the newer user-based [ACL authentication](https://redis.io/docs/management/security/acl/) option too.

Maybe we should distinguish those as:
1. `Database Password` (mentioned [here](https://redis.io/docs/management/security/#:~:text=and%20providing%20a-,database%20password,-using%20the%20requirepass)) or `Server Password` (loosely mentioned [here](https://redis.io/commands/auth/#:~:text=If%20the%20Redis%20server%20is%20password%20protected%20via%20the%20requirepass%20option.) and [here](https://redis.io/commands/auth/#:~:text=password%20in%20the%20configuration%20file%2C%20the%20server)).
4. `User Credentials` or `Database User`

I'd also be okay with leaving it as `Password` and adding support for ACL user auth later, as long as we make sure that if we print it in `op`, we prefix it with the platform name (so `Redis Password`).

Reply:
@florisvdg, thanks for the feedback! ACL authentication is definitely something on my mind and I discussed that with @AndyTitu ahead of this work as well. It just happened to slipped through when I started working/was reviewing [this particular Redis CLI documentation page](https://redis.io/docs/ui/cli/). At this stage, I am trying to tackle @hculea's [feedback](https://github.com/1Password/shell-plugins/pull/276#issuecomment-1569616959) first and once that's sorted, I'll look into ACL authentication as well.

>as long as we make sure that if we print it in op, we prefix it with the platform name

May I ask what you are referring to here? Is it the credential display whenever `op plugin init` or `op plugin inspect` is run? With that understanding, that's already the case: we show `Redis Password`.

Init phase:

<img width="1408" alt="image" src="https://github.com/1Password/shell-plugins/assets/18581859/49e76db4-43bd-47a2-80bc-c13af59ce7d6">

<img width="1405" alt="image" src="https://github.com/1Password/shell-plugins/assets/18581859/0f430017-0a1a-4dbd-88e0-d2320171f78c">

Inspect:

<img width="1119" alt="image" src="https://github.com/1Password/shell-plugins/assets/18581859/978efa4f-58e3-4567-addd-d57c5ae4c34e">

Reply:
@florisvdg, I have left some additional notes on ACL-scoped user support here: https://github.com/1Password/shell-plugins/pull/276#issuecomment-1573342063 Feedback welcome!